lib/project_types/script/commands/create.rb

```diff
@@ -26,13 +28,19 @@ def call(args, _name)
         @ctx.puts(@ctx.message('script.create.changed_dir', folder: script.name))
         @ctx.puts(@ctx.message('script.create.script_created', script_id: script.id))
       rescue StandardError => e
+        self.class.cleanup(script_name: form.name, cli_dir: cur_dir)
         UI::ErrorHandler.pretty_print_and_raise(e, failed_op: @ctx.message('script.create.error.operation_failed'))
       end
 
       def self.help
         allowed_values = Script::Layers::Application::ExtensionPoints.types.map { |type| "{{cyan:#{type}}}" }
         ShopifyCli::Context.message('script.create.help', ShopifyCli::TOOL_NAME, allowed_values.join(', '))
       end
+
+      def self.cleanup(script_name:, cli_dir:)
+        Dir.chdir(cli_dir)
+        FileUtils.rm_rf(script_name) if Dir.exist?(script_name)
```

Comment:
Since the ScriptProject class creates the directory, how do you feel about keeping cleanup logic in that class too?

Also - we should ideally be using the Context class for file operations. We have a [ticket](https://github.com/Shopify/script-service/issues/1209) in the backlog for refactoring all our code that doesn't do that.